api/test/e2e/materials/materials-get-list.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { AppModule } from 'app.module';
+import { Material } from 'modules/materials/material.entity';
+import { MaterialsModule } from 'modules/materials/materials.module';
+import { MaterialRepository } from 'modules/materials/material.repository';
+import {
+  createMaterial,
+  createSourcingLocation,
+  createSupplier,
+} from '../../entity-mocks';
+import { saveUserAndGetToken } from '../../utils/userAuth';
+import { getApp } from '../../utils/getApp';
+import { Supplier } from 'modules/suppliers/supplier.entity';
+import { LOCATION_TYPES } from 'modules/sourcing-locations/sourcing-location.entity';
+
+describe('Materials - Get the list of Materials uploaded by User with details', () => {
+  let app: INestApplication;
+  let materialRepository: MaterialRepository;
+  let jwtToken: string;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule, MaterialsModule],
+    }).compile();
+
+    materialRepository =
+      moduleFixture.get<MaterialRepository>(MaterialRepository);
+
+    app = getApp(moduleFixture);
+    await app.init();
+    jwtToken = await saveUserAndGetToken(moduleFixture, app);
+  });
+
+  afterEach(async () => {
+    await materialRepository.delete({});
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  test('Getting list of materials uploaded by users (and being part of sourcing locations) should be successful with default pagination', async () => {
```

Comment:
A few things I think can be improved in this test: 
- Add a few more materials that are not related to all the other entities, so you can validate that the endpoint filters them out correctly
- I am ok with two requests on the same test, but order them like "request1" -> "assertions1" -> "request2" -> "assertions2" as it make it easier to read.
- Your tests assume an order in the response materials ([bananas, cotton, maize]) that is not enforced anywhere. This is a nice way to get future false negatives
- You are repeating literals, like "bananas" or "maize" in your code. Those should only appear once, and then you reuse them by reference, so that it's easier to modify going forward (nitpick)